libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1540,6 +1549,9 @@ Json::Value StandardCompiler::compileYul(InputsAndSettings _inputsAndSettings)
 	if (isArtifactRequested(_inputsAndSettings.outputSelection, sourceName, contractName, "ir", wildcardMatchesExperimental))
 		output["contracts"][sourceName][contractName]["ir"] = stack.print();
 
+	if (isArtifactRequested(_inputsAndSettings.outputSelection, sourceName, contractName, "ast", wildcardMatchesExperimental))
+		output["contracts"][sourceName][contractName]["ast"] = stack.printJson();
```

Comment:
Maybe we should actually make it a file-level output for consistency with how it is for Solidity? I guess the whole Yul file represents a single contract so you could argue for doing it either way, but I think I'd go for file-level. Then we wouldn't have to make the docs for it more complicated.